language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -527,93 +553,84 @@ writePackageIdTs dir pkgId =
     T.writeFileUtf8 (dir </> "packageId.ts") $ T.unlines
       ["export default '" <> unPackageId pkgId <> "';"]
 
+-- List of texts to vector of Json strings.
+array :: [T.Text] -> Vec.Vector Json.Value
+array = Vec.fromList . map Json.String
+
+-- List of pairs of texts to a hash map of Json strings.
+fields :: [(T.Text, T.Text)] -> HMS.HashMap T.Text Json.Value
+fields fs = HMS.fromList [(a, Json.String b) | (a, b) <- fs]
+
 writeTsConfig :: FilePath -> IO ()
-writeTsConfig dir = writeFileUTF8 (dir </> "tsconfig.json") $ unlines
-    [ "{"
-    , "  \"compilerOptions\": {"
-    , "    \"target\": \"es5\","
-    , "    \"lib\": ["
-    , "      \"dom\","
-    , "      \"es2015\""
-    , "     ],"
-    , "    \"strict\": true,"
-    , "    \"noUnusedLocals\": true,"
-    , "    \"noUnusedParameters\": false,"
-    , "    \"noImplicitReturns\": true,"
-    , "    \"noFallthroughCasesInSwitch\": true,"
-    , "    \"outDir\": \"lib\","
-    , "    \"module\": \"commonjs\","
-    , "    \"declaration\": true,"
-    , "    \"sourceMap\": true"
-    , "  },"
-    , "  \"include\": [\"src/**/*.ts\"],"
-    , "}"
-    ]
+writeTsConfig dir =
+  BSL.writeFile (dir </> "tsconfig.json") $ encodePretty tsConfig
+  where
+    tsConfig :: Json.Value
+    tsConfig = Json.Object $
+      HMS.fromList [
+        ("compilerOptions", Json.Object(HMS.fromList [
+           ("target", Json.String "es5")
+         , ("lib", Json.Array (array ["dom", "es2015"]))
+         , ("strict", Json.Bool True)
+         , ("noUnusedLocals", Json.Bool True)
+         , ("noImplicitReturns", Json.Bool True)
+         , ("noFallthroughCasesInSwitch", Json.Bool True)
+         , ("outDir", Json.String "lib")
+         , ("module", Json.String "commonjs")
+         , ("declaration", Json.Bool True)
+         , ("sourceMap", Json.Bool True)]))
+      , ("include", Json.Array (array ["src/**/*.ts"]))
+      ]
```

Comment:
```suggestion
    tsConfig = Json.object
      [ "compilerOptions" Json..= Json.object
          [ "target" .= "es5"
          , "lib" .= ["dom", "es2015"]
          , "strict" .= True
          , "noUnusedLocals" .= True
          , "noImplicitReturns" .= True
          , "noFailthroughCasesInSwitch" .= True
          , "outDir" .= "lib"
          , "module" .= "commonjs"
          , "declaration" .= True
          , "sourceMap" .= True
          ]
        , "include" .= ["src/**/*.ts"]
      ]
```
Aeson provides a bunch of helpers for making this kind of code more convenient, in particular `object` and `.=`.